File: jean.cpp
#include <iostream>
#include <string>
using namespace std;

class Jean {
private:
    string codigo;
    string color;
    string talla;
    bool fueTenido;
    int cantidadTenidos;
    double precio;
    int cantidadBotones;
    int humedad; // en %
    string estadoTela; // Ej: Nueva, Desgastada, Seca

public:

    // Constructor
    Jean(string codigo, string color, string talla, bool fueTenido,
         int cantidadTenidos, double precio, int cantidadBotones,
         int humedad, string estadoTela) {

        this->codigo = codigo;
        this->color = color;
        this->talla = talla;
        this->fueTenido = fueTenido;
        this->cantidadTenidos = cantidadTenidos;
        this->precio = precio;
        this->cantidadBotones = cantidadBotones;
        this->humedad = humedad;
        this->estadoTela = estadoTela;
    }

    // Método Lavar
    void Lavar() {

        if (cantidadTenidos > 0) {
            cantidadTenidos--;

            cout << "Lavando... Cantidad de tenidos bajo a: "
                 << cantidadTenidos << endl;

            if (cantidadTenidos == 0) {
                fueTenido = false;
            }

        } else {
            cout << "El jean ya no tiene tenido." << endl;
        }

        humedad = 100;
        estadoTela = "Mojada";
    }

    // Método Secar
    void Secar() {

        if (humedad > 0) {
            humedad -= 20;

            if (humedad < 0) {
                humedad = 0;
            }

            cout << "Secando... Humedad actual: "
                 << humedad << "%" << endl;

            if (humedad == 0) {
                estadoTela = "Seca";
            }

        } else {
            cout << "El jean ya esta seco." << endl;
        }
    }

    // Método MostrarDatos
    void MostrarDatos() {

        cout << "\n--- DATOS DEL JEAN ---" << endl;
        cout << "Codigo: " << codigo << endl;
        cout << "Color: " << color << endl;
        cout << "Talla: " << talla << endl;

        cout << "Fue tenido: "
             << (fueTenido ? "Si" : "No") << endl;

        cout << "Cantidad tenidos restantes: "
             << cantidadTenidos << endl;

        cout << "Precio: $" << precio << endl;

        cout << "Cantidad botones: "
             << cantidadBotones << endl;

        cout << "Humedad: "
             << humedad << "%" << endl;

        cout << "Estado tela: "
             << estadoTela << endl;
    }
};

File: Jean.java
public class Jean {
    private String codigo;
    private String color;
    private String talla;
    private boolean fueTenido;
    private int cantidadTenidos;
    private double precio;
    private int cantidadBotones;
    private int humedad; // en %
    private String estadoTela; // Ej: Nueva, Desgastada, Seca

    public Jean(String codigo, String color, String talla, boolean fueTenido, int cantidadTenidos, double precio, int cantidadBotones, int humedad, String estadoTela) {
        this.codigo = codigo;
        this.color = color;
        this.talla = talla;
        this.fueTenido = fueTenido;
        this.cantidadTenidos = cantidadTenidos;
        this.precio = precio;
        this.cantidadBotones = cantidadBotones;
        this.humedad = humedad;
        this.estadoTela = estadoTela;
    }

    public void Lavar() {
        if (cantidadTenidos > 0) {
            cantidadTenidos--;
            System.out.println("Lavando... Cantidad de teñidos bajó a: " + cantidadTenidos);
            if (cantidadTenidos == 0) fueTenido = false;
        } else {
            System.out.println("El jean ya no tiene teñido.");
        }
        humedad = 100;
        estadoTela = "Mojada";
    }

    public void Secar() {
        if (humedad > 0) {
            humedad -= 20;
            if (humedad < 0) humedad = 0;
            System.out.println("Secando... Humedad actual: " + humedad + "%");
            if (humedad == 0) estadoTela = "Seca";
        } else {
            System.out.println("El jean ya está seco.");
        }
    }

    public void MostrarDatos() {
        System.out.println("\n--- DATOS DEL JEAN ---");
        System.out.println("Codigo: " + codigo);
        System.out.println("Color: " + color);
        System.out.println("Talla: " + talla);
        System.out.println("Fue teñido: " + (fueTenido ? "Si" : "No"));
        System.out.println("Cantidad teñidos restantes: " + cantidadTenidos);
        System.out.println("Precio: $" + precio);
        System.out.println("Cantidad botones: " + cantidadBotones);
        System.out.println("Humedad: " + humedad + "%");
        System.out.println("Estado tela: " + estadoTela);
    }
}
